src/common/catch.tcl

```diff
@@ -0,0 +1,25 @@
+# Wrap the 'catch' command so that we can decide whether to catch signals
+# or propagate them as an error. Default is to propagate; to catch them use
+# 'catch -signal ...'
+if {[info commands builtin_catch] eq {}} {
+    rename catch builtin_catch
+
+    proc catch {args} {
+        set catch_signal no
+        if {[lindex $args 0] eq "-signal"} {
+            set catch_signal yes
+            set args [lrange $args 1 end]
+        }
+        set err [uplevel 1 [list builtin_catch {*}$args]]
+        if {$err == 1 && !$catch_signal} {
```

Comment:
The documentation isn't that clear what the possible return values are, but mentions `::errorInfo` will only be set for TCL_ERROR. Then `1` would be correct.